Нужно больше функций

TheJester32 opened the pull request:
module2-task1

---
:mortar_board: [Нужно больше функций](https://up.htmlacademy.ru/javascript/28/user/2229883/tasks/7)

TheJester32 committed:
module2-task1

module2-task1

ustimenko-andrew commented on index.html:
```diff
@@ -231,6 +231,6 @@ <h2 class="success__title">Изображение успешно загруже
     <div class="img-upload__message  img-upload__message--loading">Загружаем...</div>
   </template>
-
+<script>"./js/functions.js"</script>
```
Общий комментарий к коду
1 - стиль кода нужно соблюдать https://learn.javascript.ru/coding-style
2 - используй стрелочные функции вместо обычных ( если только тебе не нужен хостинг или сохранить this )
3 - функции содержат глагол или же, если они возвращают boolean тип - могут начинаться с is / has / can и тд
4 - в идеале название функции описывает то, что она делает ( желательно максимально дословно, но нужно придерживаться баланса длина названия / смысл )

ustimenko-andrew commented on js/functions.js:
```diff
@@ -0,0 +1,34 @@
+
+isStrPalindrome('топот')
+
+function getNum(x) {
```
Функция не соответсвует ТЗ

ustimenko-andrew commented on js/functions.js:
```diff
@@ -0,0 +1,34 @@
+getNum('2023 год')
+
+function filesLocation(str, textМaxSize) {
+  if (str = str, str.length <= textМaxSize)
```
if написан не верно

ustimenko-andrew commented on js/functions.js:
```diff
@@ -0,0 +1,34 @@
+function filesLocation(str, textМaxSize) {
+  if (str = str, str.length <= textМaxSize)
+    return str;
+    alert ('Error')
```
Ни каких alert / console и etc на проверке быть не должно быть в коммите

TheJester32 committed:
module2-task1

Вроде все работает. Стрелочные функции и прочие сокращения я бы добавлял потом как шлифовку.

TheJester32 committed:
21.03 правки

Я отключил езлинт на 31 строке, потому что линтеру не нравится, что я использую str дважды. Я совершенно не понимаю как решить эту задачу, не используя str повторно.
